### ISSUE TITLE
Carousel buttons need labels

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13429]
**Describe the bug**

- Automated tests reveal missing labels in the carousel's buttons.
- While navigating with VoiceOver (or any Screen Reader), controls are missing labels and descriptions.
- These are all Top Severity bugs as per screenshots.

**Tell us the version of Spartacus**
- Library version: 4.0.0

**To Reproduce**

1. Go to Homepage
2. Open VoiceOver
3. Tab to Carousel
4. Hear the error: vocalizes "button" for each unlabeled instance.

**Expected behavior**

- Labels should be added so automated test passes.
- Labels should be meaningful so they make sense when browser with Screen Readers. E.g. Previous Slide, Current slide, Next Slide. Slide or "Group of products".

**Screenshots**
![Screen Shot 2021-08-04 at 7 14 37 PM](https://user-images.githubusercontent.com/15113219/128557484-1926f075-212c-4f77-b765-cea0e6dddfa5.png)
<img width="971" alt="Screen Shot 2021-08-04 at 7 11 44 PM" src="https://user-images.githubusercontent.com/15113219/128557491-68f20398-dadd-41c2-aa05-2ce72c5b90d6.png">
![Screen Shot 2021-08-05 at 12 04 47 PM](https://user-images.githubusercontent.com/15113219/128557636-7eedeb04-e139-439f-999c-354b0a9027ae.png)
![Screen Shot 2021-08-05 at 12 03 45 PM](https://user-images.githubusercontent.com/15113219/128557639-88f70cfd-7a59-4f43-987a-a3470eaed20a.png)
![Screen Shot 2021-08-05 at 12 03 00 PM](https://user-images.githubusercontent.com/15113219/128557643-63714f53-36c6-408a-bbd4-34dc7c7a1efc.png)
![Screen Shot 2021-08-05 at 12 01 47 PM](https://user-images.githubusercontent.com/15113219/128557646-b204cf6f-fd33-4725-a968-c95a2722bcb1.png)

**Additional context**

- This ticket doesn't aim to solve outstanding issues in the Carousel, such a those outlined in this ticket GH-9481.

**Desktop (please complete the following information):**

- OS: BigSur 11.4
- Browser Chrome Version 92.0.4515.107
- Screen Reader: VoiceOver

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.
![CarouselNextButton](https://user-images.githubusercontent.com/45409515/145873124-1a3d9d32-a571-4535-968c-e756a8dedce0.png)
![CarouselPrevButton](https://user-images.githubusercontent.com/45409515/145873137-4bdd8758-0bfc-4b2b-8840-94fb7deb607d.png)
![CarouselButton1](https://user-images.githubusercontent.com/45409515/145873145-8ad964df-2323-45e4-aa5c-ec97883b0604.png)
![CarouselButton2](https://user-images.githubusercontent.com/45409515/145873153-846239d2-41bf-4d51-8af3-56828576be2f.png)
![CarouselButton3](https://user-images.githubusercontent.com/45409515/145873173-aab8ece8-f9b0-4c0e-88eb-f9956d52fc8a.png)
![carouselButtons4](https://user-images.githubusercontent.com/45409515/145873179-65e99c63-c075-4bf4-9a7e-c25046d4d192.png)
![carouselButton5](https://user-images.githubusercontent.com/45409515/145873191-85c4be15-910b-49ae-bba1-2e06fbd89368.png)
